content/40.units_and_quantities.md

```diff
@@ -1,269 +1,682 @@
-`**` for the power function, not `^`).
+Aside from converting an array to new units as described above, there are two further ways to create
+new array and quantity objects: via a constructor, and by multiplying scalar data by a unit
+quantity.
 
-Here is a simple example of `YTArray` creation:
+#### Class constructor {#sec:class-constructor}
+
+The primary internal interface for creating new arrays and quantities is through the class
+constructor for `unyt_array`. The constructor takes three arguments. The first argument is the input
+data, which can be an integer, float, list, or array. The second argument, `input_units`, is a unit
+specification which must be a string or `Unit` instance. Last, users may optionally supply a
+`UnitRegistry` instance, which will be attached to the array. If no `UnitRegistry` is supplied, a
+default unit registry is used instead. Unit specification strings must be algebraic combinations of
+unit symbol names, using standard Python mathematical syntax (i.e. `**` for the power function, not
+`^`).
+
+Here is a simple example of `unyt_array` creation:
 
 ```python
->>> from yt.units import yt_array, YTQuantity 
->>> YTArray([1, 2, 3], `cm') 
-YTArray([1, 2, 3]) cm
->>> YTQuantity(3, `J') 
-3 J
+>>> from unyt import unyt_array, unyt_quantity 
+>>> unyt_array([1, 2, 3], 'cm') 
+unyt_array([1, 2, 3]) cm
+>>> unyt_quantity(3, 'J') 
+unyt_quantity(3, 'J') 
 ```
 
-In addition to the class constructor, we have also defined two convenience functions, `quan`, and `arr`, for quantity and array creation that are attached to the `Dataset` base class.
-These were added to syntactically simplify the creation of arrays with the UnitRegistry instance associated with a dataset.
-These functions work exactly like the `YTArray` and `YTQuantity` constructors, but pass the `UnitRegistry` instance attached to the dataset to the underlying constructor call.
-For example:
+In addition to the class constructor, we have also defined two convenience functions, `quan`, and
+`arr`, for quantity and array creation that are attached to the `Dataset` base class. These were
+added to syntactically simplify the creation of arrays with the `UnitRegistry` instance associated
+with a dataset. These functions work exactly like the `unyt_array` and `unyt_quantity` constructors,
+but pass the `UnitRegistry` instance attached to the dataset to the underlying constructor call. For
+example:
 
 ```python
 >>> import yt
 >>> ds = yt.load("Enzo_64/DD0043/data0043")
->>> ds.arr([1, 2, 3], `code_length').in_cgs() 
-YTArray([ 5.55e+26, 1.11e+27, 1.66e+27]) cm
+>>> ds.arr([1, 2, 3], 'code_length').in_cgs() 
+unyt_array([5.55e+26, 1.11e+27, 1.66e+27], 'cm')
 ```
 
-This example illustrates that the array is being created using `ds.unit_registry`, rather than the `default_unit_registry`, for which `code_length` is equivalent to `cm`.
+This example illustrates that the array is being created using `ds.unit_registry`, rather than the `default_unit_registry` in `yt`.
```

Comment:
```suggestion
The above example illustrates that the array is being created using `ds.unit_registry`, rather than the `default_unit_registry` in `yt`.
```